gdal/alg/gdalwarpkernel.cpp

```diff
@@ -6102,6 +6110,64 @@ static void GWKAverageOrModeThread( void* pData)
                         bHasFoundDensity = true;
                     }
                 }  // GRA_Average.
+                // poWK->eResample == GRA_AverageQuadratic.
+                if( nAlgo == GWKAOM_AverageQuadratic )
+                {
+                    double dfTotalReal = 0.0;
+                    double dfTotalImag = 0.0;
+                    double dfTotalWeight = 0.0;
+                    double dfR = 0.0;
+                    double dfI = 0.0;
+                    // This code adapted from GDALDownsampleChunk32R_AverageT()
+                    // in gcore/overview.cpp.
+                    for( int iSrcY = iSrcYMin; iSrcY < iSrcYMax; iSrcY++ )
+                    {
+                        const double dfWeightY = COMPUTE_WEIGHT_Y(iSrcY);
+                        iSrcOffset = iSrcXMin + static_cast<GPtrDiff_t>(iSrcY) * nSrcXSize;
+                        for( int iSrcX = iSrcXMin; iSrcX < iSrcXMax; iSrcX++, iSrcOffset++ )
+                        {
+                            if( poWK->panUnifiedSrcValid != nullptr
+                                && !(poWK->panUnifiedSrcValid[iSrcOffset>>5]
+                                     & (0x01 << (iSrcOffset & 0x1f))) )
+                            {
+                                continue;
+                            }
+
+                            if( GWKGetPixelValue(
+                                    poWK, iBand, iSrcOffset,
+                                    &dfBandDensity, &dfValueRealTmp,
+                                    &dfValueImagTmp ) &&
+                                dfBandDensity > BAND_DENSITY_THRESHOLD )
+                            {
+                                const double dfWeight = COMPUTE_WEIGHT(iSrcX, dfWeightY);
+                                dfTotalWeight += dfWeight;
+                                if (bIsComplex)
+                                {
+                                    dfTotalReal += (dfValueRealTmp * dfValueRealTmp - dfValueImagTmp *dfValueImagTmp) * dfWeight;
+                                    dfTotalImag += 2. * dfValueRealTmp * dfValueImagTmp * dfWeight;
+                                }
+                                else
+                                    dfTotalReal += dfValueRealTmp * dfWeight;
```

Comment:
There is a particular reason why you do not perform the RMS also in the case of real bands?